core/services/relay/evm/mercury/data_source.go

```diff
@@ -48,7 +46,15 @@ func (ds *datasource) Observe(ctx context.Context, repts ocrtypes.ReportTimestam
 		ds.lggr.Warnf("unable to enqueue run save for job ID %d, buffer full", ds.spec.JobID)
 	}
 
-	return ds.parse(trrs)
+	obs, err := ds.parse(trrs)
+	if err != nil {
+		return relaymercury.Observation{}, fmt.Errorf("Observe failed while parsing run results: %w", err)
+	}
+	ds.setCurrentBlock(ctx, &obs)
+
+	go collectMercuryEnhancedTelemetry(ds, &trrs, obs, repts)
```

Comment:
this seems unrelated?

Reply:
This line used to be called at the end of `ds.executeRun` which is at the beginning of this function. I pulled it out of executeRun and put it here instead. The reason being that I wanted to pass in the observation into collectMercuryEnhancedTelemetry rather than have collectMercuryEnhancedTelemetry fetch the latest block again (previously it calls `ds.parse` a second time to get values from the task run results which is fine as it doesn't involve IO, but now it could involve an eth call which I want to avoid).